Vineyard/Vineyard/ResolutionView.swift

```diff
@@ -0,0 +1,88 @@
+                    Image(systemName: resolution.isCompleted ? "checkmark.circle.fill" : "circle")
+                        .foregroundColor(resolution.isCompleted ? .green : .gray)
+                        .onTapGesture {
+                            withAnimation {
+                                viewModel.toggleResolutionAsCompleted(resolution, inGroup: group)
+                            }
+                        }
+                    Text("\(resolution.name)")
+                }
+            }.onDelete { indexSet in
+                group.resolutions.remove(atOffsets: indexSet)
+            }
+            .onMove { indexSet, index in
+                group.resolutions.move(fromOffsets: indexSet, toOffset: index)
+            }
+        }.toolbar {
+            ToolbarItem(placement: .topBarTrailing) {
+                HStack {
+                    Button {
+                       showAddItemPrompt = true
+                    } label: {
+                        Image(systemName: "plus")
+                    }.sheet(isPresented: $showAddItemPrompt, content: {
+                        VStack {
+                            TextField("Resolution Name:", text: $newResolutionName)
+                            HStack {
+                                Button {
+                                    timebound = TimeBound.day
+                                } label: {
+                                    Text("Day")
+                                }
+                                
+                                Button {
+                                    timebound = TimeBound.week
+                                } label: {
+                                    Text("Week")
+                                }
+                                
+                                Button {
+                                    timebound = TimeBound.month
+                                } label: {
+                                    Text("Month")
+                                }
+                            }
+                            TextField("Goal:", text: $goal).keyboardType(.numberPad)
+                            TextField("Frequency:", text: $frequency).keyboardType(.numberPad)
+                            
+                            Button {
+                                viewModel.addResolution(toGroup: group, resolution: Resolution(timeBound: timebound, name: newResolutionName, goal: Float(goal) ?? 0.0, freq: Int(frequency) ?? 0))
+                                showAddItemPrompt = false
+                                print("\(group.resolutions.count)")
+                            } label: {
+                                Text("SUBMIT")
+                            }
+                            
+                        }
+                    })
+                }
+            }
+        }
+    }
+}
+
+#Preview {
+    ResolutionView(group: Group(people:[Person(name: "BRuh", age: 101)], groupName: "", yearlyResolution: ""))
```

Comment:
```suggestion
    ResolutionView(group: Group(people:[Person(name: "BRuh", age: 101)], groupName: "", yearlyResolution: ""))
                .environment(ViewModel())
```

Same for the one in `ContentView.swift`